provider/cluster/kube/builder.go

```diff
@@ -288,18 +288,18 @@ func exposeExternalPort(expose *manifest.ServiceExpose) int32 {
 	return int32(expose.ExternalPort)
 }
 
+// lidNS generates a unique sha256 sum for identifying a provider's object name.
 func lidNS(lid mtypes.LeaseID) string {
-	// TODO
-	var path string
-	// path := lid.String()
-	sha := sha1.Sum([]byte(path))
+	path := mtypes.BidIDString(lid.BidID())
+	sha := sha256.Sum256([]byte(path))
```

Comment:
might run into length restrictions here (and in a lot of places), but this is necessary to expose - we need a compact representation for these things.

This doesn't need to be cryptographically secure - using a v5 uuid migtht be a good option

Reply:
Yeah, length is something I wasn't sure about. I added a test to log the length of encoded bytes. hex256 pushes it up to 64.

```
=== RUN   TestShaLengths
    TestShaLengths: builder_test.go:88: sha1: 20 sha256: 32
    TestShaLengths: builder_test.go:91: hex1: 40 hex256: 64
```